src/main/java/com/google/api/codegen/transformer/StaticLangApiMethodTransformer.java

```diff
@@ -1019,74 +874,6 @@ private RequestObjectParamView generateRequestObjectParam(
-
-    paramDoc.paramName(context.getNamer().localVarName(optionalArgsName));
-    paramDoc.typeName(context.getNamer().getOptionalArrayTypeName());
-
-    List<String> docLines = Arrays.asList("Optional.");
-
-    paramDoc.firstLine(docLines.get(0));
-    paramDoc.remainingLines(docLines.subList(1, docLines.size()));
-
-    paramDoc.arrayKeyDocs(
-        ImmutableList.<ParamDocView>builder()
-            .addAll(
-                getMethodParamDocs(
-                    context, fieldConfigs, Collections.<ParamWithSimpleDoc>emptyList()))
-            .addAll(getCallSettingsParamDocList(context))
-            .build());
-
-    return paramDoc.build();
-  }
-
-  private List<ParamDocView> getCallSettingsParamDocList(MethodTransformerContext context) {
-    List<ParamDocView> arrayKeyDocs = new ArrayList<>();
-    SimpleParamDocView.Builder retrySettingsDoc = SimpleParamDocView.newBuilder();
-    retrySettingsDoc.typeName(context.getNamer().getRetrySettingsTypeName());
-
-    Name retrySettingsName = Name.from("retry", "settings");
-    Name timeoutMillisName = Name.from("timeout", "millis");
-
-    if (context.getNamer().methodHasRetrySettings(context.getMethodConfig())) {
-      retrySettingsDoc.paramName(context.getNamer().localVarName(retrySettingsName));
-      // TODO figure out a reliable way to line-wrap comments across all languages
-      // instead of encoding it in the transformer
-      String retrySettingsDocText =
-          String.format(
-              "Retry settings to use for this call. If present, then\n%s is ignored.",
-              context.getNamer().varReference(timeoutMillisName));
-      List<String> retrySettingsDocLines = context.getNamer().getDocLines(retrySettingsDocText);
-      retrySettingsDoc.lines(retrySettingsDocLines);
-      arrayKeyDocs.add(retrySettingsDoc.build());
-    }
-
-    if (context.getNamer().methodHasTimeoutSettings(context.getMethodConfig())) {
-      SimpleParamDocView.Builder timeoutDoc = SimpleParamDocView.newBuilder();
-      timeoutDoc.typeName(
-          context.getTypeTable().getAndSaveNicknameFor(TypeRef.of(Type.TYPE_INT32)));
-      timeoutDoc.paramName(context.getNamer().localVarName(timeoutMillisName));
-      // TODO figure out a reliable way to line-wrap comments across all languages
-      // instead of encoding it in the transformer
-      String timeoutMillisDocText = "Timeout to use for this call.";
-      if (context.getNamer().methodHasRetrySettings(context.getMethodConfig())) {
-        timeoutMillisDocText +=
-            String.format(
-                " Only used if %s\nis not set.",
-                context.getNamer().varReference(retrySettingsName));
-      }
-      List<String> timeoutMillisDocLines = context.getNamer().getDocLines(timeoutMillisDocText);
-      timeoutDoc.lines(timeoutMillisDocLines);
-      arrayKeyDocs.add(timeoutDoc.build());
-    }
-
-    return arrayKeyDocs;
-  }
-
   private InitCodeContext createInitCodeContext(
```

Comment:
Can this common code between StaticLangApiMethodTransformer and DynamicLangApiMethodTransformer be extracted to the super class such as ApiMethodTransfomer?